### PR TITLE
Swagger outputs for new TemplateSpecs API version

### PR DESCRIPTION
**Pull request by TSunny007**
Changes include exception renaming and a new field in TemaplateSpecVersion object

### CODE REVIEW

**Comment by TarunSunkaraneni** [comment]
@stuartko it seems some of the code additions are from your commits

**Comment by stuartko** [comment]
@TarunSunkaraneni These were some late changes to Template Specs that didn't effect endpoints. We should be good to include the exception rename and standard SystemData updates with the SDK bump.

**Comment by TarunSunkaraneni** [comment]
@m-nash @allenjzhang This has been blocked for over a week, the corresponding swagger has been approved and merged in. Could we get sign off on this?

File: sdk/resources/Microsoft.Azure.Management.Resource/src/Generated/TemplateSpecVersionsOperations.cs [hunk 1]
```diff
@@ -253,11 +253,11 @@ public async Task<AzureOperationResponse<TemplateSpecVersion>> CreateOrUpdateWit
             string _responseContent = null;
             if ((int)_statusCode != 200 && (int)_statusCode != 201)
             {
-                var ex = new DefaultErrorResponseException(string.Format("Operation returned an invalid status code '{0}'", _statusCode));
+                var ex = new TemplateSpecsErrorException(string.Format("Operation returned an invalid status code '{0}'", _statusCode));
```
Comment (m-nash) [inline sdk/resources/Microsoft.Azure.Management.Resource/src/Generated/TemplateSpecVersionsOperations.cs:256]: Wouldn't this be a breaking change if any customer is catching DefaultErrorResponseException?

**Review by m-nash — CHANGES_REQUESTED** [review]

File: sdk/resources/Microsoft.Azure.Management.Resource/src/Generated/TemplateSpecVersionsOperations.cs (hunk 1 above)
Comment (TarunSunkaraneni) [reply sdk/resources/Microsoft.Azure.Management.Resource/src/Generated/TemplateSpecVersionsOperations.cs:256]: Echoing @stuartko: The Template Specs service is currently only an opt-in service for a select few private preview customers. We don’t anticipate that any customers will be using the SDK prior to us hitting Public Preview later this month, so there shouldn’t be any issue with the exception rename adversely impacting anyone.

**Review by m-nash — APPROVED** [review]

**Pull request merged**